### PR TITLE
refactor(providers): share the request timeout and error mapping wrapper

### PR DESCRIPTION
**Pull request by BlackHole1**
Sixth provider-level convergence PR after #441. Stacked on #447 (`refactor/entropy-b7-api-key-request`); it retargets once the stack below it lands.

## What changes

- `provider-runtime.ts` exports `runProviderRequest(options, request)`: it creates the provider timeout (`createProviderTimeout`, 30 s default), calls `request(signal)`, rethrows any `ProviderRequestError`, maps an abort or timeout to `504 "<label> request timed out"`, maps any other failure to `502 "<label> request failed: <message>"` (or `"<label> request failed"` for a non-`Error` throw), and always cleans up the timer. Four tests cover the boundary cases: passthrough of an inner `ProviderRequestError` with details, `AbortError` / `TimeoutError` / a fired timer, an `Error` and a non-`Error` failure, and a caller-supplied signal.
- 126 provider request helpers whose `try { fetch ... read body ... return } catch { rethrow guard; 504; 502 } finally { cleanup }` was exactly that shape now call `runProviderRequest({ signal, label }, async (signal) => { ... })` with the same body inside the callback. The provider label and, where present, the explicit `timeoutMs` are threaded through, so the status codes and message text are unchanged.

128 files, net -2065 lines.

## What was converted, and what was not

The audit found 329 catch blocks with this timeout/abort ladder across 319 files, in 56 semantic variants. Only sites that are byte-equivalent under the helper were converted, selected by script:

- the block has the `instanceof ProviderRequestError` rethrow guard, the 504 message is exactly `"<label> request timed out"`, the 502 pair is exactly `` `<label> request failed: ${error.message}` `` / `"<label> request failed"` with the same label, no raw `error` is passed as `details`, and the catch has no extra branches (149 blocks);
- and `createProviderTimeout` is the statement right before the `try`, the try body refers to the timeout only as `timeout.signal`, and nothing follows the `finally` inside the function (126 of those 149).

Left alone on purpose: the 23 canonical blocks where the timeout is created earlier or code follows the `finally` (converting would move the timeout window; adyntel's credential-validation block also uses its own messages), the 40-odd sites that time-box only the fetch and read the body after `finally`, the 46 without a rethrow guard (converting would change how guarded-fetch and SSRF-block messages surface), the 72 that attach the raw error as `details`, the branded or non-canonical messages, and one-offs with extra branches.

## Wire behavior

None. For every converted site the helper produces the same `ProviderRequestError` (status, message, no details) as the deleted block, and `ProviderRequestError` thrown inside the callback passes through untouched. The evaluation order is the same: the timeout is created before the request, and the message is built from a label that is a string literal at every converted site (no template interpolation moved earlier).

## Verification

- `npm run fix-check`, `npx vitest run` (163 files, 1558 tests), `npm run build --workspace web`, `git diff --check`: green.
- `npm run generate:catalog`: all 1451 `catalog/apps/*.json` byte-identical to `origin/main`.
- The classifier re-run on the result finds 0 remaining convertible blocks; the only surviving `createProviderTimeout` + `isAbortLikeError` ladders are the non-canonical ones listed above.
- Differential run: for 16 converted providers the module at the base commit and the converted module were driven through their real exported handlers with the same injected fetcher over abort, timeout, `Error`, non-`Error`, inner `ProviderRequestError` with details, a fired timer, 200/4xx/5xx bodies, parent-signal aborts and a mid-body stream failure: 261 comparisons of status, message, details and resolved value, 0 differences, 0 leaked timers. A structural check over all 126 sites confirms each callback body equals the old try body modulo `timeout.signal` -> `signal`, with no outer `signal` shadowed, no `this` / `arguments`, and the try as the function's last statement.

### CODE REVIEW

File: src/providers/semantic_scholar/executors.ts [hunk 1]
```diff
@@ -347,14 +346,12 @@ async function requestSemanticScholarJson(input: {
...
-  const timeoutHandle = createProviderTimeout(undefined);
-
-  try {
+  return runProviderRequest({ label: "Semantic Scholar" }, async (signal) => {
     const response = await input.fetcher(buildSemanticScholarUrl(input), {
       method: input.method,
       headers: buildSemanticScholarHeaders(input),
       body: input.method === "POST" ? JSON.stringify(input.body ?? {}) : undefined,
-      signal: timeoutHandle.signal,
+      signal,
```
[the remark this answers is not in the document]
Comment (BlackHole1) [reply src/providers/semantic_scholar/executors.ts:354]: @coderabbitai This is intentional. These helpers already called `createProviderTimeout(undefined, ...)` before this conversion, so they never forwarded a caller AbortSignal. The PR only extracts the timeout and error-mapping wrapper and keeps wire behavior unchanged. Threading a caller signal through request helpers that never accepted one is a separate behavior change, not a missed conversion.
